matrix_sdk_appservice/src/lib.rs

```diff
@@ -173,34 +175,31 @@ impl Deref for AppserviceRegistration {
     }
 }
 
-async fn client_session_with_login_restore(
-    client: &Client,
-    registration: &AppserviceRegistration,
-    localpart: impl AsRef<str> + Into<Box<str>>,
-    server_name: &ServerNameBox,
-) -> Result<()> {
-    let session = Session {
-        access_token: registration.as_token.clone(),
-        user_id: UserId::parse_with_server_name(localpart, server_name)?,
-        device_id: DeviceId::new(),
-    };
-    client.restore_login(session).await?;
-
-    Ok(())
-}
+type Localpart = String;
+
+/// The main appservice user is the `sender_localpart` from the given
+/// [`AppserviceRegistration`]
+///
+/// Dummy type for shared documentation
+#[allow(dead_code)]
+pub type MainAppserviceUser = ();
```

Comment:
I wonder whether there's a more elegant way for rust docs glossaries :thinking:

Reply:
Well, could probably work with simply a doc-only module and linking there

Reply:
Yeah I have been planing to do a matrix-sdk tutorial module for a while, though haven't managed to get around to it yet.